aten/src/ATen/native/cpu/ReduceUtils.h

```diff
@@ -142,6 +152,37 @@ inline void update(scalar_t* out, scalar_t* data, int64_t K) {
       K);
 }
 
+template <typename scalar_t, ReductionType reduce,
+          typename std::enable_if_t<!is_reduced_floating_point_v<scalar_t>, int> = 0>
+inline void _update(scalar_t* src_data, at::opmath_type<scalar_t>* buffer_ptr, int64_t col, int64_t K) {
+  update<scalar_t, reduce>(buffer_ptr, src_data + col * K, K);
+}
+
+template <typename scalar_t, ReductionType reduce,
+          typename std::enable_if_t<is_reduced_floating_point_v<scalar_t>, int> = 0>
+inline void _update(scalar_t* src_data, at::opmath_type<scalar_t>* buffer_ptr, int64_t col, int64_t K) {
+  using opmath_t = at::opmath_type<scalar_t>;
+  using Vec = vec::Vectorized<scalar_t>;
+  using fVec = vec::Vectorized<opmath_t>;
+  int64_t k = 0;
+  constexpr int64_t kVecSize = Vec::size();
+  constexpr int64_t kfVecSize = fVec::size();
+  for (k = 0; k < K - (K % kVecSize); k += kVecSize) {
```

Comment:
it is easier to add a overload for

template <typename scalar_t, typename accumut>
map {
  static_assert(); // check scalar_t is bfloat16 or float16
};

template <scalar_t>
void map<scalar_t, scalar_t> { vec::map(); }